src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -303,7 +303,7 @@ protected function prepareResponse($request, Exception $e)
         }
 
         if (! $this->isHttpException($e)) {
-            $e = new HttpException(500, $e->getMessage());
+            $e = new HttpException($e->getCode(), $e->getMessage());
```

Comment:
This is not the http code, this is the exception code. Exception codes are nothing to do with HTTP.